src/cpp/statistics/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -0,0 +1,407 @@
+    {
+        std::string topic_name = writer->get_topic()->get_name();
+        disable_statistics_datawriter(topic_name);
+    }
+
+    // Delete builtin_publisher
+    delete_publisher(builtin_publisher_);
+}
+
+const std::string DomainParticipantImpl::transform_topic_name_alias(
+        const std::string& topic) noexcept
+{
+    std::string topic_name;
+    if (HISTORY_LATENCY_TOPIC_ALIAS == topic)
+    {
+        topic_name = HISTORY_LATENCY_TOPIC;
+    }
+    else if (NETWORK_LATENCY_TOPIC_ALIAS == topic)
+    {
+        topic_name = NETWORK_LATENCY_TOPIC;
+    }
+    else if (PUBLICATION_THROUGHPUT_TOPIC_ALIAS == topic)
+    {
+        topic_name = PUBLICATION_THROUGHPUT_TOPIC;
+    }
+    else if (SUBSCRIPTION_THROUGHPUT_TOPIC_ALIAS == topic)
+    {
+        topic_name = SUBSCRIPTION_THROUGHPUT_TOPIC;
+    }
+    else if (RTPS_SENT_TOPIC_ALIAS == topic)
+    {
+        topic_name = RTPS_SENT_TOPIC;
+    }
+    else if (RTPS_LOST_TOPIC_ALIAS == topic)
+    {
+        topic_name = RTPS_LOST_TOPIC;
+    }
+    else if (RESENT_DATAS_TOPIC_ALIAS == topic)
+    {
+        topic_name = RESENT_DATAS_TOPIC;
+    }
+    else if (HEARTBEAT_COUNT_TOPIC_ALIAS == topic)
+    {
+        topic_name = HEARTBEAT_COUNT_TOPIC;
+    }
+    else if (ACKNACK_COUNT_TOPIC_ALIAS == topic)
+    {
+        topic_name = ACKNACK_COUNT_TOPIC;
+    }
+    else if (NACKFRAG_COUNT_TOPIC_ALIAS == topic)
+    {
+        topic_name = NACKFRAG_COUNT_TOPIC;
+    }
+    else if (GAP_COUNT_TOPIC_ALIAS == topic)
+    {
+        topic_name = GAP_COUNT_TOPIC;
+    }
+    else if (DATA_COUNT_TOPIC_ALIAS == topic)
+    {
+        topic_name = DATA_COUNT_TOPIC;
+    }
+    else if (PDP_PACKETS_TOPIC_ALIAS == topic)
+    {
+        topic_name = PDP_PACKETS_TOPIC;
+    }
+    else if (EDP_PACKETS_TOPIC_ALIAS == topic)
+    {
+        topic_name = EDP_PACKETS_TOPIC;
+    }
+    else if (DISCOVERY_TOPIC_ALIAS == topic)
+    {
+        topic_name = DISCOVERY_TOPIC;
+    }
+    else if (SAMPLE_DATAS_TOPIC_ALIAS == topic)
+    {
+        topic_name = SAMPLE_DATAS_TOPIC;
+    }
+    else if (PHYSICAL_DATA_TOPIC_ALIAS == topic)
+    {
+        topic_name = PHYSICAL_DATA_TOPIC;
+    }
+    else
+    {
+        topic_name = topic;
+    }
+    return topic_name;
+}
+
+bool DomainParticipantImpl::check_statistics_topic_name(
+        const std::string& topic) noexcept
+{
+    if (HISTORY_LATENCY_TOPIC != topic && NETWORK_LATENCY_TOPIC != topic && PUBLICATION_THROUGHPUT_TOPIC != topic &&
+            SUBSCRIPTION_THROUGHPUT_TOPIC != topic && RTPS_SENT_TOPIC != topic && RTPS_LOST_TOPIC != topic &&
+            RESENT_DATAS_TOPIC != topic && HEARTBEAT_COUNT_TOPIC != topic && ACKNACK_COUNT_TOPIC != topic &&
+            NACKFRAG_COUNT_TOPIC != topic && GAP_COUNT_TOPIC != topic && DATA_COUNT_TOPIC != topic &&
+            PDP_PACKETS_TOPIC != topic && EDP_PACKETS_TOPIC != topic && DISCOVERY_TOPIC != topic &&
+            SAMPLE_DATAS_TOPIC != topic && PHYSICAL_DATA_TOPIC != topic)
```

Comment:
I know it's linter friendly but ... 🙄

Reply:
This method is unified with `transform_topic_name_alias` in #1924.